src/tools/compiletest/src/header.rs

```diff
@@ -843,6 +844,21 @@ pub fn extract_llvm_version(version: &str) -> Option<u32> {
     Some(version)
 }
 
+pub fn extract_llvm_version_from_binary(binary_path: Option<String>) -> Option<u32> {
+    let binary_path = binary_path?;
```

Comment:
Maybe put this `?` in the caller instead? So this function can take a `&str` input.

Reply:
Good point. :+1: